### PR TITLE
Record approvals when posting shares; Delete ephemeral key after provision

### PR DESCRIPTION
**Pull request by emostov**
- Submit an approval when posting share and record the approval for audit trail.
- Get rid of `QuroumSet` and instead have `ManifestSet` and `ShareSet`. `ManifestSet` is a set of keys and threshold for approving the manifest. `ShareSet` is a set keys that can post share, and by proxy approve the attestation flow.
- Delete the ephemeral key once the quorum key is provisioned
- Change set from https://github.com/tkhq/qos/pull/134
- Change set from #135 

TODO:
- [x] Fix docs and add section to explain Manifest Set vs Share Set
- [x] Add tests in provision
- [x] Get e2e tests to work

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,11 +68,19 @@ An asymmetric key used to uniquely authenticate and encrypt data. This key shoul
...
 
-#### Quorum Set
+#### Quorum Sets
 
-The collection of members whom each hold shares of the Quorum Key and can authorize certain QOS actions. A threshold of these members shares is required to reconstruct the Quorum Key
+There are two typs of quorum sets:
+
+##### Manifest Set
+
+The collection of members whom can approve a manifest.ma
```
Comment (emostov) [inline README.md]:
```suggestion
The collection of members whom can approve a manifest.
```

File: integration/tests/boot.rs [hunk 2]
```diff
@@ -101,15 +101,24 @@ async fn boot_e2e() {
...
 		.iter()
 		.map(|m| QuorumMember {
 			alias: m.setup_member.alias.clone(),
 			pub_key: m.public_personal_key.clone(),
 		})
 		.collect();
-	quorum_set_members.sort();
+	manifest_set_members.sort();
+
+	let share_set_members = manifest_set_members
```
Comment (emostov) [inline integration/tests/boot.rs]: These are the same as manifest set to make the test easier, but don't strictly have to be.

File: qos_client/src/cli/mod.rs [hunk 3]
```diff
@@ -32,7 +32,7 @@
 //!
 //! ### Quorum Key Generation
 //!
```
Comment (emostov) [inline qos_client/src/cli/mod.rs]: Giving up on updating the docs in this PR - leaving for a follow up

File: README.md [hunk 4]
```diff
@@ -68,17 +68,25 @@ An asymmetric key used to uniquely authenticate and encrypt data. This key shoul
...
-An entity that is a member of the Quorum Set and holds a share of the Quorum Key.
+An entity that is a member of the Manifest Set and holds a share of the Quorum Key.
 
-#### Quorum Set
+#### Quorum Sets
 
-The collection of members whom each hold shares of the Quorum Key and can authorize certain QOS actions. A threshold of these members shares is required to reconstruct the Quorum Key
+There are two typs of quorum sets:
+
+##### Manifest Set
```
Comment (jack-kearney) [inline README.md]: Think we should standardize on `manifest set` or `release set` or something else.

File: src/qos_client/src/cli/services.rs [hunk 5]
```diff
@@ -338,7 +348,16 @@ pub(crate) fn generate_manifest<P: AsRef<Path>>(args: GenerateManifestArgs<P>) {
 			args: pivot_args,
 		},
 		quorum_key: genesis_output.quorum_key,
-		quorum_set: QuorumSet { threshold: genesis_output.threshold, members },
+		// TODO: the members of the manifest set should not be from the genesis
```
Comment (jack-kearney) [inline src/qos_client/src/cli/services.rs:351]: Are you going to address this in a follow-on PR?

Comment (emostov) [reply src/qos_client/src/cli/services.rs:351]: yee

**Pull request merged**